app/Services/Model/Imp/AbstractPublishService.php

```diff
@@ -0,0 +1,305 @@
+                        (
+                            sprintf
+                            (
+                                "You can't publish on this time frame, it conflicts with event id %s.",
+                                $c_event->getId()
+                            )
+                        );
+                    }
+                }
+
+                // check speakers collisions
+                if ($publishable_event instanceof IPublishableEvent && $c_event instanceof IPublishableEvent &&
+                    $publishable_event->getId() != $c_event->getId()) {
+                    if (!$eventType->isAllowsSpeakerEventCollision()) {
+                        foreach ($publishable_event->getSpeakers() as $current_speaker) {
+                            foreach ($c_event->getSpeakers() as $c_speaker) {
+                                if (intval($c_speaker->getId()) === intval($current_speaker->getId())) {
+                                    throw new ValidationException
+                                    (
+                                        sprintf
+                                        (
+                                            "You can't publish Event %s (%s) on this timeframe, speaker %s its present in room %s at this time.",
+                                            $publishable_event->getTitle(),
+                                            $publishable_event->getId(),
+                                            $current_speaker->getFullName(),
+                                            $c_event->getLocationName()
+                                        )
+                                    );
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param Summit $summit
+     * @param SummitEvent $event
+     * @param array $data
+     * @param bool $from_scheduled_event
+     * @return SummitEvent
+     * @throws \Exception
+     */
+    protected function publishCurrentEvent(Summit $summit, SummitEvent $event, array $data, bool $from_scheduled_event = false): SummitEvent
+    {
+        return $this->tx_service->transaction(function () use ($summit, $event, $data, $from_scheduled_event) {
+
+            if (!$event->hasType())
+                throw new EntityNotFoundException("Event type its not assigned to event id {$event->getIdentifier()}!");
+
+            $type = $event->getType();
+
+            if (is_null($event->getSummit()))
+                throw new EntityNotFoundException("Summit its not assigned to event id {$event->getIdentifier()}!");
+
+            if ($event->getSummit()->getIdentifier() !== $summit->getIdentifier())
+                throw new ValidationException(
+                    "Event {$event->getIdentifier()} does not belongs to summit id {$summit->getIdentifier()}");
+
+            $this->updateEventDates($data, $summit, $event);
```

Comment:
this expect a IPublishableEvent but SummitEvent does not implement this interface

Reply:
done